sparse/_common.py

```diff
@@ -0,0 +1,387 @@
+import numpy as np
+
+from ._coo import (
+    tensordot,
+    dot,
+    matmul,
+    triu,
+    tril,
+    where,
+    nansum,
+    nanmean,
+    nanprod,
+    nanmin,
+    nanmax,
+    nanreduce,
+    roll,
+    kron,
+    argwhere,
+    isposinf,
+    isneginf,
+    result_type,
+    diagonal,
+    diagonalize,
+    elemwise,
+    as_coo,
+)
+
+
+def stack(arrays, axis=0, compressed_axes=None):
+    """
+    Stack the input arrays along the given dimension.
+
+    Parameters
+    ----------
+    arrays : Iterable[SparseArray]
+        The input arrays to stack.
+    axis : int, optional
+        The axis along which to stack the input arrays.
+    compressed_axes : iterable, optional
+        The axes to compress if returning a GCXS array.
+
+    Returns
+    -------
+    COO or GCXS
+        The output stacked array.
+
+    Raises
+    ------
+    ValueError
+        If all elements of :code:`arrays` don't have the same fill-value.
+
+    See Also
+    --------
+    numpy.stack : NumPy equivalent function
+    """
+    from ._compressed import GCXS
+
+    gcxs_arrays = 0
+    for arr in arrays:
+        if isinstance(arr, GCXS):
+            gcxs_arrays += 1
+    if gcxs_arrays > len(arrays) / 2:
```

Comment:
I would suggest picking a firm rule about which sparse type has higher precedence rather than going with the majority. I think this would be easier to understand and would result in more predictable behavior.

Reply:
I agree here. I would think, for now, dispatch to `COO` if there's any `COO` array present.